### PR TITLE
Demo refactor: make all demos available as standalone examples

### PR DESCRIPTION
**Pull request by Pessimistress**
As promised, the 100k line PR.

### CODE REVIEW

File: examples/plot/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      options: {
+        objectAssign: 'Object.assign'
+      }
+    }]
+  },
+
+  resolve: {
+    alias: {
+      // From mapbox-gl-js README. Required for non-browserify bundlers (e.g. webpack):
+      'deck.gl': resolve('./node_modules/deck.gl'),
```
Comment (ibgreen) [inline examples/plot/webpack.config.js]: ditto

File: examples/scatterplot/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+This is a minimal standalone version of the TripsLayer example
```
Comment (ibgreen) [inline examples/scatterplot/README.md]: Copy paste

File: examples/scatterplot/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+### Usage
+Copy the content of this folder to your project. Run
+```
+npm install
+npm start
+```
+
+### Data format
+Sample data is stored in the `data` folder. To use your own data, checkout
+the [documentation of TripsLayer](./trips-layer/README.md).
```
Comment (ibgreen) [inline examples/scatterplot/README.md]: copy paste

File: examples/screen-grid/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+This is a minimal standalone version of the TripsLayer example
```
Comment (ibgreen) [inline examples/screen-grid/README.md]: copy paste here and at bottom

File: examples/screen-grid/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      options: {
+        objectAssign: 'Object.assign'
+      }
+    }]
+  },
+
+  resolve: {
+    alias: {
+      // From mapbox-gl-js README. Required for non-browserify bundlers (e.g. webpack):
+      'deck.gl': resolve('./node_modules/deck.gl'),
```
Comment (ibgreen) [inline examples/screen-grid/webpack.config.js]: ditto

File: examples/icon/data/location-icon-mapping.json [hunk 6]
```diff
@@ -0,0 +1,154 @@
+{
```
Comment (Pessimistress) [inline examples/icon/data/location-icon-mapping.json]: This is a duplicate. Copying static files from external directories when building demo can be messy.

File: examples/icon/deckgl-overlay.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      return null;
+    }
+
+    const z = Math.floor(viewport.zoom);
+    const size = showCluster ? 1 : Math.min(Math.pow(1.5, viewport.zoom - 10), 1);
+
+    const layer = new IconLayer({
+      id: 'icon',
+      data,
+      pickable: this.props.onHover || this.props.onClick,
```
Comment (Pessimistress) [inline examples/icon/deckgl-overlay.js]: Sublayers need to be pickable for composite layer to be picked, even they don't have callbacks themselves.

**Pull request merged**